### DISCUSSION TITLE
Installation of eulerpi fails because of amici, what can I do about it?

### DISCUSSION THREAD

**Discussion by shoepfl** [Systems-Theory-in-Systems-Biology/EPI#37] [Q&A] [answered]
Amici is used for models written in the Systems Biology Markup Language (SBML). It needs SWIG, a CBLAS compatible BLAS library, a C++ and C compiler.

Further hdf5 can be used for parallelization.

**Comment by shoepfl** [comment, marked as answer]
The following instruction can be used to install amici in an virtual conda environment:

conda create --name <Enviroment-Name>

conda activate <Enviroment-Name>


conda install gcc_linux-64 gxx_linux-64 swig blas libopenblas
conda install -c conda-forge blas openblas

export CONDA_DIR=$CONDA_PREFIX

conda install hdf5
conda install pip

pip install --no-cache amici

afterwards eulerpi can be installed via `pip install eulerpi`.

**Comment by shoepfl** [comment]
Amici is not needed for users that do not work with SBML models.

You can install the eulerpi package without its dependencies via `pip install eulerpi --no-deps` and afterward only install the dependencies you need.

**Reply by kaiserls** [reply]
Installation of eulerpi without amici/sbml is now, since version 0.6.0, the default: `pip install eulerpi`
For sbml support you can execute `pip install eulerpi[sbml]`
